habitat/datasets/rearrange/samplers.py

```diff
@@ -276,32 +294,58 @@ def sample_placement(
                     logger.info(
                         f"Successfully sampled (snapped) object placement in {num_placement_tries} tries."
                     )
+                    if not self.is_accessible(sim, new_object):
+                        continue
                     return new_object
 
             elif not new_object.contact_test():
                 logger.info(
                     f"Successfully sampled object placement in {num_placement_tries} tries."
                 )
+                if not self.is_accessible(sim, new_object):
+                    continue
                 return new_object
 
         # if num_placement_tries > self.max_placement_attempts:
         sim.get_rigid_object_manager().remove_object_by_handle(
             new_object.handle
         )
-        logger.info(
-            f"Failed to sample object placement in {self.max_placement_attempts} tries."
+        logger.warning(
+            f"Failed to sample {object_handle} placement on {receptacle.name} in {self.max_placement_attempts} tries."
         )
+
         return None
 
+    def is_accessible(self, sim, new_object):
```

Comment:
Make this method private?

Reply:
Also note that there are edge cases where this logic will result in false negatives. For example, spawning objects on a corner bookshelf where the agent may fit behind the shelf, so the closest snap point for all objects on the back half of the bookshelf would be rejected, despite being accessible.  

Maybe we should leave a TODO to clean this up or document the limitation?

Reply:
I agree there can be edge cases. I documented this. In reality, I do not think the solution comes here but in a per-task filtering step. Checking for these edge cases seems task specific and it's hard to add every edge case to the generic episode generation code.